src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,201 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null || list.size() == 0) {
+            return;
+        }
+        linkAllLast(list);
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> x = node(index);
+        T oldVal = x.value;
+        x.value = value;
+        return oldVal;
     }
```

Comment:
The variable names `x` and `oldVal` are not very descriptive. Consider renaming `x` to `node` or `currentNode`, and `oldVal` to `oldValue`.